packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -95,24 +98,20 @@ contract PluginRepo is
         address _pluginSetup,
         bytes calldata _contentURI
     ) external auth(address(this), CREATE_VERSION_PERMISSION_ID) {
-        // Check if `_pluginSetup` is a `PluginSetup` contract
-        if (!Address.isContract(_pluginSetup)) {
-            revert InvalidContractAddress({invalidContract: _pluginSetup});
+        // In a case where _pluginSetup doesn't contain supportsInterface,
+        // but contains fallback, that doesn't return anything(most cases)
+        // the below approach aims to still return custom error which not possible with try/catch..
+        // NOTE: also checks if _pluginSetup is a contract and reverts if not.
+        bytes memory data = _pluginSetup.functionCall(
+            abi.encodeWithSelector(ERC165.supportsInterface.selector, type(PluginSetup).interfaceId)
+        );
+
+        // NOTE: if data contains 32 bytes that can't be decoded with uint256
+        // it reverts with solidity's ambigious error.
+        if (data.length != 32 || abi.decode(data, (uint256)) != 1) {
+            revert InvalidPluginSetupInterface({invalidPluginSetup: _pluginSetup});
         }
 
-        // TODO: uncommment
-        // try
-        //     PluginSetup(_pluginSetup).supportsInterface(
-        //         PluginSetup.PLUGIN_FACTORY_INTERFACE_ID
-        //     )
-        // returns (bool result) {
-        //     if (!result) {
-        //         revert InvalidPluginSetupInterface({invalidPluginSetup: _pluginSetup});
-        //     }
-        // } catch {
-        //     revert InvalidPluginSetupContract({invalidPluginSetup: _pluginSetup});
-        // }
```

Comment:
Why ceasing to check this?

Reply:
This had one problem which I didn't like (the problem existed more in EVM level limitation from solidity with try/catch !